src/ucp/core/ucp_context.h

```diff
@@ -275,6 +275,12 @@ typedef struct ucp_context {
     ucp_tl_md_t                   *tl_mds;    /* Memory domain resources */
     ucp_md_index_t                num_mds;    /* Number of memory domains */
 
+    /* Map of memory domains that have valid memory keys for host memory
+       allocated with ucp_mem_mmap_common().
+       It's initialized on first access. */
+    int                           alloc_md_map_initialized;
+    ucp_md_map_t                  alloc_md_map;
```

Comment:
why do we need it?

Reply:
To cache the results of `ucp_memh_register`. Like it was before https://github.com/openucx/ucx/pull/8392.

Reply:
Removed.